Initial provider review - changes

ewilde opened the pull request:
This PR references the issues raised in #27

ewilde committed:
fix: add golint to build pipeline and fix source issues

ewilde committed:
fix: use goimport instead of gofmt

ewilde committed:
build: update make file to use goimports

ewilde committed:
bug: Check 404 as well as 403 when reading a bucket resource

ewilde committed:
bug: Don't export config struct

ewilde committed:
bug: Remove unused function validateValueFunc

kevholditch commented on scripts/goimportscheck.sh:
```diff
@@ -0,0 +1,13 @@
+if [[ -n ${goimports_files} ]]; then
+    echo 'goimports needs running on the following files:'
+    echo "${goimports_files}"
+    echo "You can use the command: \`make fmt\` to reformat code."
```
message is wrong here should be `make goimports`

ewilde committed:
fix: use TypeSet where order is not important

ewilde committed:
update missing docs and remove step_token resource

ewilde committed:
Merge branch 'master' into provider-review

paultyng commented on runscope/data_source_runscope_integration_test.go:
```diff
@@ -4,21 +4,22 @@ import (
 	"fmt"
 	"testing"
 
```
not a big deal here, `goimports` sometimes doesn't quite get the whitespacing right as it errs on the side of including your import groups, but I'd group `os` with the other stdlib packages normally, but won't hold up process on this.

paultyng commented on runscope/data_source_runscope_integration_test.go:
```diff
@@ -57,7 +58,7 @@ data "runscope_integration" "by_type" {
 func TestAccDataSourceRunscopeIntegration_Filter(t *testing.T) {
 
-	teamId := os.Getenv("RUNSCOPE_TEAM_ID")
+	teamIS := os.Getenv("RUNSCOPE_TEAM_ID")
```
this may be a typo?

paultyng commented on runscope/provider_test.go:
```diff
@@ -4,11 +4,12 @@ import (
 	"os"
 	"testing"
 
```
Similar issue here with extra whitespace, but not a big deal.

ewilde committed:
fix: code review feedback

1. Fix issues with import statements generated by goimports
2. Add comment at top of resource_renscope_test_.go as to reason
3. Fix variable type teamID

Signed-off-by: Edward Wilde <ewilde@gmail.com>

ewilde commented:
@paultyng thanks for the review, i've updated where you have commented. What are the next steps?

paultyng commented on website/docs/d/integrations.html.markdown:
```diff
@@ -0,0 +1,36 @@
+
+# runscope\_integration
+
+Use this data source to list all of your [integration]s(https://www.runscope.com/docs/api/integrations)
```
Not sure if this `s` needs to be inside the brackets? or the URL next to the brackets?

paultyng commented on website/docs/d/integrations.html.markdown:
```diff
@@ -0,0 +1,36 @@
+* `filter` - (Optional) Filter to reduce the list of integrations returned.
+
+Variables (`filter`) supports the following:
+
```
Missing some copy here, if you are not ready to add content, can just remove this paragraph.

paultyng commented:
Sorry had one more minor catch in the markdown. Will follow up in email about next steps.

cgriggs01 commented:
Hi @ewilde,

PR looks great, when do you plan to merge?

Best,
Chris

ewilde committed:
Merge branch 'master' into provider-review

ewilde committed:
docs: update integration docs to make attribute reference corrections

Signed-off-by: Edward Wilde <ewilde@gmail.com>

ewilde commented:
@cgriggs01 thanks for the feedback i've updated the integration docs and merged the PR
